diff --git a/plugins/linkit_search/file.class.php b/plugins/linkit_search/file.class.php
index cd77c00..271bd5b 100644
--- a/plugins/linkit_search/file.class.php
+++ b/plugins/linkit_search/file.class.php
@@ -78,6 +78,23 @@ class LinkitSearchPluginFile extends LinkitSearchPluginEntity {
   }
 
   /**
+   * Overrides LinkitSearchPluginEntity::createPath().
+   *
+   * If 'Direct download' is enabled, make the link point to the file entity
+   * download endpoint.
+   */
+  function createPath($entity) {
+    if (isset($this->conf['direct_download']) && $this->conf['direct_download']) {
+      $uri = file_entity_download_uri($entity);
+      // Process the uri with the insert pluing.
+      return linkit_get_insert_plugin_processed_path($this->profile, $uri['path'], $uri['options']);
+    }
+    else {
+      return parent::createPath($entity);
+    }
+  }
+
+  /**
    * Overrides LinkitSearchPluginEntity::getQueryInstance().
    */
   function getQueryInstance() {
@@ -105,6 +122,12 @@ class LinkitSearchPluginFile extends LinkitSearchPluginEntity {
       '#default_value' => isset($this->conf['group_by_scheme']) ? $this->conf['group_by_scheme'] : '',
     );
 
+    $form['entity:file']['direct_download'] = array(
+      '#title' => t('Direct download'),
+      '#type' => 'checkbox',
+      '#default_value' => isset($this->conf['direct_download']) ? $this->conf['direct_download'] : '',
+    );
+
     $image_extra_info_options = array(
       'thumbnail' => t('Show thumbnails <em>(using the image style !linkit_thumb_link)</em>', array('!linkit_thumb_link' => l(t('linkit_thumb'), 'admin/config/media/image-styles/edit/linkit_thumb'))),
       'dimensions' => t('Show pixel dimensions'),
