diff --git a/modules/contrib/feeds_xls/feeds_xls.module b/modules/contrib/feeds_xls/feeds_xls.module
index 7ee784a..a7bf99e 100644
--- a/modules/contrib/feeds_xls/feeds_xls.module
+++ b/modules/contrib/feeds_xls/feeds_xls.module
@@ -188,13 +188,16 @@ function feeds_xls_feeds_processor_targets_alter(&$targets, $entity_type, $bundl
  * Callback for setting date fields
  */
 function feeds_xls_feeds_set_target_date($source, $entity, $target, $value){
+  $info = field_info_field($target);
+  $mapping['language'] = entity_metadata_field_get_language($entity->type, $entity, $info['field_name']);
+  
   if(strpos(get_class($source->importer->parser), 'FeedsExcelParser') !== FALSE && is_numeric($value)){
     $userDate = PHPExcel_Shared_Date::ExcelToPHPObject($value);
     $date = new DateTime($userDate->format('Y-m-d H:i:s'), new DateTimeZone(drupal_get_user_timezone()));
     $date->setTimezone(new DateTimeZone('UTC'));
     $value = $date->format('Y-m-d H:i:s');
   }
-  return date_feeds_set_target($source, $entity, $target, $value);
+  return date_feeds_set_target($source, $entity, $target, $value, $mapping);
 }
 
 /**
