diff --git a/print.pages.inc b/print.pages.inc
index a7bd734..c3e288d 100644
--- a/print.pages.inc
+++ b/print.pages.inc
@@ -183,7 +183,7 @@ function _print_var_generator($node, $query = NULL, $message = NULL, $cid = NULL
   $print['title'] = check_plain($node->title);
   $print['head'] = drupal_get_html_head();
   if ($print_html_sendtoprinter) {
-    drupal_add_js('misc/drupal.js', array('weight' => JS_LIBRARY));
+    drupal_add_js($GLOBALS['base_url'] . 'misc/drupal.js', array('weight' => JS_LIBRARY));
   }
   $print['scripts'] = drupal_get_js();
   $print['footer_scripts'] = drupal_get_js('footer');
